src/QsCompiler/Core/Dependencies.fs

```diff
@@ -36,7 +36,8 @@ type BuiltIn = {
     /// These should be non-Generic callables only.
     static member RewriteStepDependencies =
         ImmutableHashSet.Create (
-            BuiltIn.RangeReverse.FullName
+            BuiltIn.RangeReverse.FullName,
+            BuiltIn.Length.FullName
```

Comment:
Because `Length` is generic and not intrinsic, it will cause problems later on if it is not monomorphized. Having it here means that the compilation will contain generics all the way through to the end.

Reply:
I think Length is intrinsic, but intrinsic in the sense of being `body intrinsic` and not in the sense of being part of `Microsoft.Quantum.Intrinsic`. It's defined that way in [Core.qs](https://github.com/microsoft/qsharp-runtime/blob/master/src/Simulation/QsharpCore/Core.qs#L58)

Reply:
Ah you are right, it is intrinsic. But that means that it doesn't need to be in this list as it will be picked up with all the other intrinsics at the [beginning of the monomorphization transformation](https://github.com/microsoft/qsharp-compiler/blob/beheim/bugFixes/src/QsCompiler/Transformations/Monomorphization.cs#L54).

Reply:
Oh, I think I understand better now. So because Length is intrinsic it will already be skipped and doesn't need to be explicitly listed for skipping again. That seems reasonable to me, and probably explains why it works without this.

Reply:
Given the name "RewriteStepDependencies" I think it makes sense to list all dependencies, independent on whether they are intrinsic or not. That also has the advantage that the monomorphization validation properly fails if there is a generic intrinsic dependency that is not properly handled.